tests/unit/test_download.py

```diff
@@ -157,6 +157,43 @@ def test_unpack_http_url_bad_downloaded_checksum(mock_unpack_file):
         rmtree(download_dir)
 
 
+def test_download_http_url_with_directory_traversal(data):
+    """
+    It should download file to temp_sub_dir
+    """
+    mock_url = 'http://www.example.com/whatever.tgz'
+    contents = b'downloaded'
+    link = Link(mock_url)
+
+    session = Mock()
+    resp = MockResponse(contents)
+    resp.url = mock_url
+    resp.headers = {
+        # set random avoid mimetypes.guess_extension guess ext
+        'content-type': 'random',
+        'content-disposition': 'attachment;filename="../out_dir_file"'
+    }
+    session.get.return_value = resp
+
+    temp_dir = mkdtemp()
```

Comment:
You can use the pytest `tmpdir` fixture so you won't need to manually create and delete a temp dir.